index.html

```diff
@@ -2866,32 +2836,57 @@ <h3><dfn>Writing or pushing content</dfn></h3>
         <a href="https://www.w3.org/2001/tag/doc/URNsAndRegistries-50.html">
         W3C TAG recommendations</a> to avoid using URNs in the Web.
         Therefore, Web NFC does not use the URN prefix neither when reading
-        or writing <a>NDEF messages</a>, but requires the leading colon ('`:`')
-        before the type name.
+        or writing <a>NDEF messages</a>.
       </p>
       <p>
-        To <dfn>validate external type</dfn> given a |input:USVString| run
+        To <dfn>validate external type</dfn> given |input:USVString|, run
         these steps:
       </p>
       <ol class=algorithm id="validate-external-type">
         <li>
           If |input:string| is not a {{USVString}} or it is empty, or its
-          length exceeds 255 bytes, or it contains white spaces,
-          return `false`.
+          length exceeds 255 bytes, return `false`.
         </li>
         <li>
-          If |input| does not start with colon ('`:`'), return |input|.
-          <p class="issue">
-            Validating domain name needs an algorithm.
-          </p>
+          Find the last occurrence of `U+003A` (`:`) in |input|.
+          If not found, return `false`.
+        </li>
+        <li>
+          Let |domain| be the part of |input| preceding the last colon,
+          and |type| be the part following the last colon.
+        </li>
+        <li>
+          Run the following sub-steps of
+          <a href="https://url.spec.whatwg.org/#host-parsing">host parsing</a>
+          on |domain|:
+          <ul>
+            <li>
+              Let |asciiDomain| be the result of running
+              <a href="https://url.spec.whatwg.org/#concept-domain-to-ascii">
+              domain to ASCII</a> on |domain|.
```

Comment:
What are we actually writing? Are we writing the UTF-8 version of urls always? Maybe this deserves a note or so somewhere

Reply:
We are not writing anything, just checking if app (user?) provided string qualifies for ext type name.

Reply:
Not here, but when we are writing the type to the NDEF we are writing something :) - I assume as UTF-8

Reply:
Let's track that in separate issue. This is for removing effective domain.

Reply:
Sure, please file